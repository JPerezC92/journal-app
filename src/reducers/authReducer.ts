import { createSlice, PayloadAction } from "@reduxjs/toolkit";

export interface AuthState {
  isLoggedIn: boolean;
  displayName: string;
  uuid: string;
}

const initialState: AuthState = {
  isLoggedIn: false,
} as AuthState;

export const authSlice = createSlice({
  name: "[Auth]",
  initialState,
  reducers: {
    login: (_, action: PayloadAction<AuthState>): AuthState => action.payload,
    logout: () => initialState,
  },
});

export const { login, logout } = authSlice.actions;

const authReducer = authSlice.reducer;

export default authReducer;
