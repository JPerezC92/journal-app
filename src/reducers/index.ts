export * from "./authReducer";
export * from "./uiReducer";
export * from "./notesReducer";
