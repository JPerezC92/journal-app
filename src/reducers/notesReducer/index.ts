export * from "./notesReducer";
export * from "./notesThunks";
