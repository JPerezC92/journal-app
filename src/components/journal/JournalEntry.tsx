import { format } from "date-fns";
import { useDispatch } from "react-redux";
import { Note, notesActions } from "../../reducers";

const JournalEntry = (props: { note: Note }) => {
  const dispatch = useDispatch();
  const { title, body, date, imageUrl } = props.note;

  const handleEntryClick = () => {
    dispatch(notesActions.setNoteActive(props.note));
  };

  return (
    <div
      className="journal__entry pointer animate__animated animate__fadeIn animate__faster"
      onClick={handleEntryClick}
    >
      {imageUrl && (
        <div
          className="journal__entry-picture"
          style={{
            backgroundSize: "cover",
            backgroundImage: `url(${imageUrl})`,
          }}
        ></div>
      )}

      {/* <div
        className="journal__entry-picture"
        style={{
          backgroundSize: "cover",
          backgroundImage:
            "url(https://cdn.pixabay.com/photo/2015/04/23/22/00/tree-736885__480.jpg)",
        }}
      ></div> */}

      <div className="journal__entry-body">
        <p className="journal__entry-title">{title}</p>
        <p className="journal__entry-content">{body}</p>
      </div>
      <div className="journal__entry-date-box">
        <span>{format(date, "eeee")}</span>
        <h4>{format(date, "do")}</h4>
      </div>
    </div>
  );
};

export default JournalEntry;
