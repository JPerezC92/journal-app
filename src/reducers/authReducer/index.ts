import authReducer from "./authReducer";

export * from "./authReducer";
export * from "./authReducerTypes";
export * from "./authThunks";

export default authReducer;
