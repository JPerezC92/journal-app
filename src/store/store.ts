import { configureStore } from "@reduxjs/toolkit";
import { uiReducer, authReducer, notesReducer } from "../reducers";

export const reducers = {
  authReducer,
  uiReducer,
  notesReducer,
};

export const store = configureStore({
  reducer: reducers,
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
