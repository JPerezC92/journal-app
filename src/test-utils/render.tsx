import { configureStore } from "@reduxjs/toolkit";
import { render, RenderOptions } from "@testing-library/react";
import { Provider } from "react-redux";
import { uiReducer, authReducer, notesReducer } from "../reducers";

const customRender = (
  ui: React.ReactElement<any, string | React.JSXElementConstructor<any>>,
  options?: RenderOptions<
    typeof import("/home/philip/projects/journal-app/node_modules/@testing-library/dom/types/queries"),
    HTMLElement
  >
) => {
  const store = configureStore({
    reducer: {
      authReducer,
      uiReducer,
      notesReducer,
    },
  });
  const reduxStore: React.FC = ({ children }) => {
    return <Provider store={store}>{children}</Provider>;
  };

  return { component: render(ui, { wrapper: reduxStore, ...options }), store };
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
