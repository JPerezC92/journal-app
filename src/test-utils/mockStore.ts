import {
  AnyAction,
  CombinedState,
  combineReducers,
  configureStore,
  createSlice,
} from "@reduxjs/toolkit";
import { reducers } from "../store/store";

const storeCleanSlice = createSlice({
  name: "store",
  initialState: {},
  reducers: {
    clean: () => {},
  },
});

export const storeCleanActions = storeCleanSlice.actions;
const storeCleanReducer = storeCleanSlice.reducer;
const combinedReducers = combineReducers({ ...reducers, storeCleanReducer });

const testReducers = (
  state: CombinedState<any> | undefined,
  action: AnyAction
) => {
  if (action.type === "store/clean") {
    state = undefined;
  }
  return combinedReducers(state, action);
};

export const mockStore = configureStore({
  reducer: testReducers,
});

export type MockRootState = ReturnType<typeof mockStore.getState>;
export type MockAppDispatch = typeof mockStore.dispatch;
