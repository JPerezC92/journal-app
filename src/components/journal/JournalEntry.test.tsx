import { fireEvent, render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { format } from "date-fns";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import { Note, notesActions } from "../../reducers";
import { store } from "../../store/store";
import { mockStore } from "../../test-utils/mockStore";
import JournalEntry from "./JournalEntry";

describe("Test on <JournalEntry />", () => {
  const noteMock: Note = {
    id: "1",
    title: "Test title",
    body: "Test body",
    date: new Date().getDate(),
    imageUrl:
      "https://res.cloudinary.com/jperezc92/image/upload/v1627759301/ezxhoydkvqaidpcf6loq.png",
  };

  test("should match the snapshot correctly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <JournalEntry note={noteMock} />
          </MemoryRouter>
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  test("should call notesActions.setNoteActive", () => {
    jest.spyOn(notesActions, "setNoteActive");

    const component = render(
      <Provider store={mockStore}>
        <MemoryRouter>
          <JournalEntry note={noteMock} />
        </MemoryRouter>
      </Provider>
    );

    const entry = component.getByRole("listitem");

    fireEvent.click(entry);

    expect(notesActions.setNoteActive).toHaveBeenCalledTimes(1);
    expect(notesActions.setNoteActive).toHaveBeenCalledWith(noteMock);
  });

  test("should contain the note information", async () => {
    const component = render(
      <Provider store={mockStore}>
        <MemoryRouter>
          <JournalEntry note={noteMock} />
        </MemoryRouter>
      </Provider>
    );
    const entry = component.getByRole("listitem");

    expect(await component.findByRole("img")).toBeInTheDocument();
    expect(entry).toHaveTextContent(noteMock.title);
    expect(entry).toHaveTextContent(noteMock.body);
    expect(entry).toHaveTextContent(format(noteMock.date, "eeee"));
    expect(entry).toHaveTextContent(format(noteMock.date, "do"));
  });
});
