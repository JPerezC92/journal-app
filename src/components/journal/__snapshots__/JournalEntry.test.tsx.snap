// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <JournalEntry /> should match the snapshot correctly 1`] = `
<div
  className="journal__entry pointer animate__animated animate__fadeIn animate__faster"
  onClick={[Function]}
  role="listitem"
>
  <div
    className="journal__entry-picture"
    role="img"
    style={
      Object {
        "backgroundImage": "url(https://res.cloudinary.com/jperezc92/image/upload/v1627759301/ezxhoydkvqaidpcf6loq.png)",
        "backgroundSize": "cover",
      }
    }
  />
  <div
    className="journal__entry-body"
  >
    <p
      className="journal__entry-title"
    >
      Test title
    </p>
    <p
      className="journal__entry-content"
    >
      Test body
    </p>
  </div>
  <div
    className="journal__entry-date-box"
  >
    <span>
      Wednesday
    </span>
    <h4>
      31st
    </h4>
  </div>
</div>
`;
